internal/repository/sync.go

```diff
@@ -62,8 +67,33 @@ func Clone(ctx context.Context, remoteURL, dir, initialBranch string) (*Reposito
 		return nil, errors.Join(ErrCloningRepository, err)
 	}
 
+	sort.Slice(expectedRootKeys, func(i, j int) bool {
+		return expectedRootKeys[i].KeyID < expectedRootKeys[j].KeyID
+	})
+
 	repository := &Repository{r: r}
 
+	slog.Debug("Verifying if root keys are expected root keys...")
+	if len(expectedRootKeys) > 0 {
```

Comment:
```suggestion
	if len(expectedRootKeys) > 0 {
	  slog.Debug("Verifying if root keys are expected root keys...")
```
should be switched as it's only verifying if pubkeys are passed

Reply:
Fixed